services/harmony/app/models/workflow-steps.ts

```diff
@@ -264,3 +283,54 @@ export async function decrementWorkItemCount(tx: Transaction, jobID, stepIndex):
     .where({ jobID, stepIndex })
     .decrement('workItemCount');
 }
+
+/**
+ * Determine whether or not the workflow step is complete and set its `is_complete` column
+ * to `true` if so.
+ *
+ * @param tx - the database transaction
+ * @param jobID - the job ID
+ * @param step - the current workflow step
+ * @returns a Promise containing a boolean that indicates whether or not the step is complete
+ */
+export async function updateIsComplete(tx: Transaction, jobID: string, numInputGranules: number, step: WorkflowStep): Promise<boolean> {
+
+  let isComplete = false;
+
+  const { stepIndex } = step;
+
+  if (step.is_sequential) {
+    const completedCount = await workItemCountForStep(tx, jobID, stepIndex, COMPLETED_WORK_ITEM_STATUSES);
+    const expectedCount =  Math.ceil(numInputGranules / env.cmrMaxPageSize);
+    isComplete = completedCount == expectedCount;
+
+  } else {
+    let prevStepComplete = true;
+    if (stepIndex > 1) {
```

Comment:
Seems like there might an issue with this logic if the first workflow step in a chain is not sequential. I don't see any way we could mark the step as completed in this case (probably need to add a test for this scenario - I'm thinking Giovanni might fit this case unless you are marking the giovanni-adapter as sequential).

Reply:
This is not an issue

Reply:
I missed line 308 is marking prevStepComplete = true which prevents the issue I was thinking of.